CHT/KappaEffective.C

```diff
@@ -47,136 +47,32 @@ scalar preciceAdapter::CHT::KappaEff_Compressible::getAt(int i)
-        {
-            adapterInfo
-            (
-                "Cannot find the density in preciceDict/CHT using the name " +
-                nameRho_,
-                "error"
-            );
-        }
-        else
-        {
-            DEBUG(adapterInfo("  rho = " + std::to_string(rho_.value())));
-        }
-
-        // Read the heat capacity
-        if (!CHTDict.readIfPresent<dimensionedScalar>(nameCp_, Cp_))
-        {
-            adapterInfo
-            (
-                "Cannot find the heat capacity in preciceDict/CHT using the name " +
-                nameCp_,
-                "error"
-            );
-        }
-        else
-        {
-            DEBUG(adapterInfo("  Cp = " + std::to_string(Cp_.value())));
-        }
 }
 
 void preciceAdapter::CHT::KappaEff_Incompressible::extract(uint patchID, bool meshConnectivity)
 {
-    // Compute kappaEff_ from the turbulence model, using alpha and Prandl
-
-    // Get the laminar viscosity from the turbulence model
-    // TODO: Do we really need turbulence at the end?
-    const scalarField & nu
-    (
-        turbulence_.nu() ().boundaryField()[patchID]
-    );
-
-    // Compute the effective thermal diffusivity
-    // (alphaEff = alpha + alphat = nu / Pr + nut / Prt)
-
-    scalarField alphaEff;
-
-    // Does the turbulent thermal diffusivity exist in the object registry?
-    if (mesh_.foundObject<volScalarField>(nameAlphat_))
-    {
-        const scalarField & alphat
-        (
-            mesh_.lookupObject<volScalarField>(nameAlphat_).boundaryField()[patchID]
-        );
-
-        alphaEff = nu / Pr_.value() + alphat;
-    }
-    else
-    {
-        WarningInFunction
-            << "The object alphat does not exist. "
-            << "An incompressible solver should create it explicitly "
-            << "(e.g. in the createFields.H)."
-            << "Assuming only the laminar part of the thermal diffusivity."
-            << nl;
-
-        alphaEff = nu / Pr_.value();
-    }
-
-    // Compute the effective thermal conductivity and store it in a temp variable
-    scalarField kappaEff_temp
-    (
-        alphaEff * rho_.value() * Cp_.value()
-    );
-
-    if(meshConnectivity)
+    if (meshConnectivity)
     {
         //Create an Interpolation object at the boundary Field
         primitivePatchInterpolation patchInterpolator(mesh_.boundaryMesh()[patchID]);
 
-        //Interpolate kappaEff_ from centers to nodes, if desired
-        kappaEff_= patchInterpolator.faceToPointInterpolate(kappaEff_temp);
+        //Interpolate kappaEff_ from centers to nodes
+        kappaEff_ = patchInterpolator.faceToPointInterpolate(thermophysicalTransportModel_.kappaEff()().boundaryField()[patchID]);
     }
     else
     {
-        // if no interpolation
-        kappaEff_ = kappaEff_temp;
+        // Extract kappaEff_ from the thermophysical transport model
+        kappaEff_ = thermophysicalTransportModel_.kappaEff()().boundaryField()[patchID];
```

Comment:
With this change, we don't need to distinguish between incompressible and compressible solvers anymore. This raises new possibilities for simplifying the code, but also new questions on how to deal with different OpenFOAM versions (see #32 #128 ).